apps/showcase/react-connect-query/src/main.tsx

```diff
@@ -36,35 +43,63 @@ const locationQuery = createJsonQuery({
   },
 });
 
-connectQuery({
-  // Right after characterQuery is successfully finished
-  source: characterQuery,
-  // get location URL from it
-  fn(character) {
-    return { params: { locationUrl: character.location.url } };
+let newConnectQuery: any;
+
+// Character with location inside
+const characterInfoQuery = newConnectQuery(
+  {
+    // We need information from characterQuery before locationQuery execution
+    source: characterQuery,
+    // get location URL from it
+    // we can pass queryParams if we want
+    fn(character: Character, queryParams: CharacterQueryParams) {
+      return { params: { locationUrl: character.location.url } };
+    },
+    // and start locationQuery with it
+    target: locationQuery,
   },
-  // and start locationQuery with it
-  target: locationQuery,
-});
+  (character: Character, location: Location) => ({
```

Comment:
What is the difference between this function and `fn` in the previous argument?

It looks like a mix of `connectQuery` and `combineQuery` 🤔 In my mind, this mix can be hard to understand. Could we split it back?

Reply:
`fn` is for passing results from `source` to `target`. And that's function is for preparing the result of a whole new combined query.
I agree, it looks a little bit complicated. I'm trying to solve two issues:
1. Non-explicit control of connection with `enable` field. Usage of a new combined query makes things more explicit.
2. Usage of data from few requests as one whole thing.

Second issue could be solved by `combineQuery(q1, q2, q3,..., fn)`. It looks nice and pretty simple. For solving first issue I can propose another option: return start event from `connectQuery`

```javascript
const requestUserInfo = connectQuery({
  target: userIdQuery,
  fn: ...,
  source: userInfoQuery
})
```

Reply:
We can add stop event too.

```
const {start: requestUserInfo, abort: abortUserInfoRequest} = connectQuery({
  target: userIdQuery,
  fn: ...,
  source: userInfoQuery
})```

Reply:
Returning of event approach doesn't solve the problem of big query chain `q1 -> q2 -> q3`. If we return a new query from `connectQuery` we can solve it like this:
```
const q1q2 = connectQuery({source: q1, target: q2}, prepareDataForQ3);
const q2q3 = connectQuery({source: q1q2, target: q3});
``` 
I don't think is possible to remove `prepareDataForQ3` (but you haven't to use it every time). Also, we can use it in object form

```javascript
connectQuery({
  source,
  fn,
  target,
  prepareResult/resultFn/resFn/outFn
})
```